### DISCUSSION TITLE
How Android uses IKEv1/IPSec PSK connections

### DISCUSSION THREAD

**Discussion by gityms123** [strongswan/strongswan#1047] [Q&A]
I have added the following configuration in the ipsec.conf file, the way iOS works fine with IPsec，But Android can't connect all the time, I've also tried the configuration examples provided by https://wiki.strongswan.org/projects/strongswan/wiki/UsableExamples without success.
Does Android need any other configuration?

conn IPsec-Xauth-PSK
    keyexchange=ikev1
    ike=aes256-sha256-modp2048,3des-sha1-modp2048,aes256-sha1-modp2048!
    esp=aes256-sha256,3des-sha1,aes256-sha1!
    left=%defaultroute
    leftauth=psk
    leftsubnet=0.0.0.0/0
    right=%any
    rightauth=psk
    rightauth2=xauth
    rightsourceip=10.31.1.0/24
    rightsubnet=0.0.0.0/0
    auto=add

**Comment by tobiasbrunner** [comment]
No idea, don't have any experience with IKEv1 on Android. Consider using IKEv2 instead.

**Reply by gityms123** [reply]
But mobile phones below Android 12 seem to only support the ikev1 method
